### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by Outspending** [Minestom/Minestom#1913] [Help]
- How would one save playerdata? Like their position (for loading in again), ect
- I'm working with blockstates and how would i do this efficiently 

Thanks for your time :heart:

**Comment by Tofaa2** [comment]
1) To save player data you need a database, then on join and on quit (or on player's data change) save it to the database.
2) Could you elaborate more as to what you are talking about? What are you working on

**Comment by MasterBroNetwork** [comment]
Not sure what you mean by 2, But for saving player data, You can use a database as @Tofaa2 said or use something like JSON file storage, However, A database would be your best bet here if you have a lot of data to work with and want to store them in an easily accessible manner.

**Reply by Outspending** [reply]
Ah okay, i was just wondering if there was something built into minestom but i guess not. For blockstates i was wondering how i could use them easily since right now all directional, waterlogged, ect is just 1 state and cannot be changed
